scripts/cosmosdb_migration.py

```diff
@@ -0,0 +1,162 @@
+"""
+A migration script to migrate data from CosmosDB to a new format.
+The old schema:
+id: str
+entra_oid: str
+title: str
+timestamp: int
+answers: list of 2-item list of str, dict
+
+The new schema has two item types in the same container:
+For session items:
+id: str
+session_id: str
+entra_oid: str
+title: str
+timestamp: int
+type: str (always "session")
+version: str (always "cosmosdb-v2")
+
+For message_pair items:
+id: str
+session_id: str
+entra_oid: str
+type: str (always "message_pair")
+version: str (always "cosmosdb-v2")
+question: str
+response: dict
+"""
+
+import os
+
+from azure.cosmos.aio import CosmosClient
+from azure.identity.aio import AzureDeveloperCliCredential
+
+from load_azd_env import load_azd_env
+
+
+class CosmosDBMigrator:
+    """
+    Migrator class for CosmosDB data migration.
+    """
+
+    def __init__(self, cosmos_account, database_name, credential=None):
+        """
+        Initialize the migrator with CosmosDB account and database.
+
+        Args:
+            cosmos_account: CosmosDB account name
+            database_name: Database name
+            credential: Azure credential, defaults to AzureDeveloperCliCredential
+        """
+        self.cosmos_account = cosmos_account
+        self.database_name = database_name
+        self.credential = credential or AzureDeveloperCliCredential()
+        self.client = None
+        self.database = None
+        self.old_container = None
+        self.new_container = None
+
+    async def connect(self):
+        """
+        Connect to CosmosDB and initialize containers.
+        """
+        self.client = CosmosClient(
+            url=f"https://{self.cosmos_account}.documents.azure.com:443/", credential=self.credential
+        )
+        self.database = self.client.get_database_client(self.database_name)
+        self.old_container = self.database.get_container_client("chat-history")
+        self.new_container = self.database.get_container_client("chat-history-v2")
+        try:
+            await self.old_container.read()
+        except Exception:
+            raise ValueError(f"Old container {self.old_container.id} does not exist")
+        try:
+            await self.new_container.read()
+        except Exception:
+            raise ValueError(f"New container {self.new_container.id} does not exist")
```

Comment:
Does migration script need to create the container?

Reply:
No, the Bicep creates the containers, it has all the right configuration for the partitions and hash and such.